src/jit/importer.cpp

```diff
@@ -13637,7 +13636,14 @@ bool Compiler::impReturnInstruction(BasicBlock *block, int prefixFlags, OPCODE &
                     bool restoreType = false;
                     if ((op2->OperGet() == GT_CALL) && (info.compRetType == TYP_STRUCT))
                     {
-                        noway_assert(op2->TypeGet() == TYP_STRUCT); // If not, then impFixupStructReturnType() must have touched it!
+                        // If the op2 type is not TYP_STRUCT, then the impFixupStructReturnType has changed it.
+                        // For System V a single eightbyte struct's type could be normalized to the type of the single eightbyte.
```

Comment:
In my opinion, the System V normalization should be happening at the same time that the other return type normalization is done (i.e. for other targets). In that case, this would be where the type of op2 would be changed from TYP_STRUCT. However, I think that will require some significant refactoring, so is probably best left for future.